ZBus linking problem with C++ on latest main branch

I have this issue caused by C++ mangling (I believe):

`ld.bfd.exe: app/libapp.a(TemperatureEvent.cpp.obj):(._zbus_channel_observation.static.temperature_chan00_+0x4): undefined reference to 'computing_th_temp_lis'`

Here is the code:

```
ZBUS_CHAN_DEFINE(temperature_chan,
                 TemperatureEvent,
                 nullptr,
                 nullptr,
                 ZBUS_OBSERVERS(computing_th_temp_lis),
                 ZBUS_MSG_INIT()
);
```

and listender is defined in another cpp file:

`ZBUS_LISTENER_DEFINE(computing_th_temp_lis, on_zbus_temperature_callback);`

It works on 3.7 branch.

--- a/include/zephyr/zbus/zbus.h
+++ b/include/zephyr/zbus/zbus.h
@@ -415,7 +415,7 @@ struct zbus_channel_observation {
 			.priority = ZBUS_MIN_THREAD_PRIORITY,                 \
 		))                                                            \
 	};                                                                    \
-	const STRUCT_SECTION_ITERABLE(zbus_observer, _name) = {               \
+	_ZBUS_CPP_EXTERN const STRUCT_SECTION_ITERABLE(zbus_observer, _name) = {  \
 		ZBUS_OBSERVER_NAME_INIT(_name) /* Name field */               \
 		.type = ZBUS_OBSERVER_SUBSCRIBER_TYPE,                        \
 		.data = &_CONCAT(_zbus_obs_data_, _name),                     \
@@ -457,7 +457,7 @@ struct zbus_channel_observation {
 			.priority = ZBUS_MIN_THREAD_PRIORITY,                                      \
 		))                                                                                 \
 	};                                                                                         \
-	const STRUCT_SECTION_ITERABLE(zbus_observer, _name) = {                                    \
+	_ZBUS_CPP_EXTERN const STRUCT_SECTION_ITERABLE(zbus_observer, _name) = {                   \
 		ZBUS_OBSERVER_NAME_INIT(_name) /* Name field */                                    \
 		.type = ZBUS_OBSERVER_LISTENER_TYPE,                                               \
 		.data = &_CONCAT(_zbus_obs_data_, _name),                                          \
@@ -497,7 +497,7 @@ struct zbus_channel_observation {
 			.priority = ZBUS_MIN_THREAD_PRIORITY,                 \
 		))                                                            \
 	};                                                                    \
-	const STRUCT_SECTION_ITERABLE(zbus_observer, _name) = {               \
+	_ZBUS_CPP_EXTERN const STRUCT_SECTION_ITERABLE(zbus_observer, _name) = {  \
 		ZBUS_OBSERVER_NAME_INIT(_name) /* Name field */               \
 		.type = ZBUS_OBSERVER_MSG_SUBSCRIBER_TYPE,                    \
 		.data = &_CONCAT(_zbus_obs_data_, _name),                     \

